extensions/ql-vscode/src/databases/database-fetcher.ts

```diff
@@ -42,605 +41,566 @@ import { createFilenameFromString } from "../common/filenames";
-    },
-    progress,
-    cli,
-    makeSelected,
-    addSourceArchiveFolder,
-  );
-}
+    } = result;
 
-/**
- * Imports a database from a local archive.
- *
- * @param databaseUrl the file url of the archive to import
- * @param databaseManager the DatabaseManager
- * @param storagePath where to store the unzipped database.
- * @param cli the CodeQL CLI server
- */
-export async function importArchiveDatabase(
-  commandManager: AppCommandManager,
-  databaseUrl: string,
-  databaseManager: DatabaseManager,
-  storagePath: string,
-  progress: ProgressCallback,
-  cli: CodeQLCliServer,
-): Promise<DatabaseItem | undefined> {
-  try {
-    const item = await databaseArchiveFetcher(
+    return this.downloadGitHubDatabaseFromUrl(
+      databaseUrl,
+      databaseId,
+      databaseCreatedAt,
+      commitOid,
+      owner,
+      name,
+      octokit,
+      progress,
+      makeSelected,
+      addSourceArchiveFolder,
+    );
+  }
+
+  public async downloadGitHubDatabaseFromUrl(
+    databaseUrl: string,
+    databaseId: number,
+    databaseCreatedAt: string,
+    commitOid: string | null,
+    owner: string,
+    name: string,
+    octokit: Octokit,
+    progress: ProgressCallback,
+    makeSelected = true,
+    addSourceArchiveFolder = true,
+  ): Promise<DatabaseItem | undefined> {
+    /**
+     * The 'token' property of the token object returned by `octokit.auth()`.
+     * The object is undocumented, but looks something like this:
+     * {
+     *   token: 'xxxx',
+     *   tokenType: 'oauth',
+     *   type: 'token',
+     * }
+     * We only need the actual token string.
+     */
+    const octokitToken = ((await octokit.auth()) as { token: string })?.token;
+    return await this.databaseArchiveFetcher(
       databaseUrl,
-      {},
-      databaseManager,
-      storagePath,
-      undefined,
       {
-        type: "archive",
-        path: databaseUrl,
+        Accept: "application/zip",
+        Authorization: octokitToken ? `Bearer ${octokitToken}` : "",
+      },
+      `${owner}/${name}`,
+      {
+        type: "github",
+        repository: `${owner}/${name}`,
+        databaseId,
+        databaseCreatedAt,
+        commitOid,
       },
       progress,
-      cli,
+      makeSelected,
+      addSourceArchiveFolder,
     );
-    if (item) {
-      await commandManager.execute("codeQLDatabases.focus");
-      void showAndLogInformationMessage(
-        extLogger,
-        "Database unzipped and imported successfully.",
-      );
-    }
-    return item;
-  } catch (e) {
-    if (getErrorMessage(e).includes("unexpected end of file")) {
-      throw new Error(
-        "Database is corrupt or too large. Try unzipping outside of VS Code and importing the unzipped folder instead.",
+  }
+
+  /**
+   * Imports a database from a local archive.
+   *
+   * @param databaseUrl the file url of the archive to import
+   */
+  public async importArchiveDatabase(
+    databaseUrl: string,
+    progress: ProgressCallback,
```

Comment:
This isn't from your PR, but a JSDoc description is missing for `progress`. Could you add this here?